### PR TITLE
fix(module-type-aliases): move @type packages to dependencies

### LINKED ISSUE #5463: Module '"@docusaurus/router"' has no exported member 'matchPath'
### Prerequisites

- [x] I'm using the latest version of Docusaurus.
- [x] I have tried the `npm run clear` or `yarn clear` command.
- [x] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io
- [x] I have read the console error message carefully (if applicable)

### Description

I'm using `docusaurus` in a TypeScript project. When we run `tsc -b` to build the project, we get this error:

```
> tsc -b

node_modules/.pnpm/@docusaurus+plugin-content-docs@2.0.0-beta.5_react-dom@17.0.2+react@17.0.2/node_modules/@docusaurus/plugin-content-docs/src/client/docsClientUtils.ts:8:9 - error TS2305: Module '"@docusaurus/router"' has no exported member 'matchPath'.

8 import {matchPath} from '@docusaurus/router';
          ~~~~~~~~~
```

`@docusaurus/plugin-content-docs` [imports](https://unpkg.com/@docusaurus/plugin-content-docs@2.0.0-beta.5/src/client/docsClientUtils.ts) `matchPath` from `@docusaurus/router`. But I couldn't find `@docusaurus/router` in my `node_modules` nor https://www.npmjs.com/. 

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues)?

yes

## Steps to reproduce

Use https://new.docusaurus.io to create a CodeSandbox reproducible demo of the bug.

(Write your steps here:)

1. Install `@docusaurus/plugin-content-docs@2.0.0-beta.5`
1. Use TypeScript to compile the project

## Expected behavior

 
No error.

If `@docusaurus/router` doesn't mean to be an NPM package, then maybe docusaurus should export `.d.ts` files instead of raw `.ts` files as types.

## Actual behavior

Got error.

## Your environment

- Public source code: N/A
- Public site URL: N/A 
- Docusaurus version used: `@docusaurus/plugin-content-docs@2.0.0-beta.5`
- Environment name and version (e.g. Chrome 78.0.3904.108, Node.js 10.17.0): Node.js v16.7.0, TypeScript 4.4.2, Pnpm 6.14.5
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): macOS 11.5

## Reproducible demo

N/A

### PR DESCRIPTION
**Pull request by Josh-Cena**
## Motivation

Should fix #5463. It seems we can also update the docs since installing `@docusaurus/module-type-aliases` would install all the `@types/*` dependencies so they no longer need to be installed separately?

Also removed `index.js` file from `docusaurus-types` as it doesn't really make sense.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

File: website/docs/typescript-support.md [hunk 1]
```diff
@@ -17,10 +17,10 @@ Below are some guides on how to migrate an existing project to TypeScript.
 
 ## Setup {#setup}
 
-To start using TypeScript, add `@docusaurus/module-type-aliases` and some `@types` dependencies to your project:
+To start using TypeScript, add `@docusaurus/module-type-aliases` and the base TS config to your project:
 
 ```bash npm2yarn
-npm install --save-dev typescript @docusaurus/module-type-aliases @types/react @types/react-router-dom @types/react-helmet @tsconfig/docusaurus
```
Comment (slorber) [inline website/docs/typescript-support.md:23]: should we remove those from the TS template as well?

Are you sure it works? I don't think we have any e2e test on this yet

File: packages/docusaurus-types/src/index.js [hunk 2]
```diff
@@ -1,10 +0,0 @@
-/**
- * Copyright (c) Facebook, Inc. and its affiliates.
- *
- * This source code is licensed under the MIT license found in the
- * LICENSE file in the root directory of this source tree.
- */
-
-/* eslint-disable */
-'use strict';
-Object.defineProperty(exports, '__esModule', {value: true});
```
Comment (slorber) [inline packages/docusaurus-types/src/index.js:10]: I have no idea what this does, any idea? :D

**Review by slorber — CHANGES_REQUESTED** [review]
LGTM thanks, just missing ts template update?

File: packages/docusaurus-types/src/index.js (hunk 2 above)
Comment (Josh-Cena) [reply packages/docusaurus-types/src/index.js:10]: This is how the compiled output of a `.ts` file with no value exports looks like. I guess it's just someone didn't know a package doesn't necessarily need a `main` entry?

File: website/docs/typescript-support.md (hunk 1 above)
Comment (Josh-Cena) [reply website/docs/typescript-support.md:23]: `dependencies` will be installed transitively but `devDependencies` don't, so afaik it should work. I'll update the template and check it out

Comment (slorber) [reply website/docs/typescript-support.md:23]: Thanks, let me know if the TS template still works

Comment (Josh-Cena) [reply website/docs/typescript-support.md:23]: Taking longer than I thought🧐 Tried to set up a verdaccio registry locally, but when I `npm_config_registry=http://localhost:4873 yarn install`, I still get the real `2.0.0-beta.6` version on npm

Comment (Josh-Cena) [reply website/docs/typescript-support.md:23]: @slorber Tested, it's working🎉

Comment (slorber) [reply website/docs/typescript-support.md:23]: Thanks 👍 

FYI I added a readme in init package with simpler local testing instructions without Verdaccio:

---

# `@docusaurus/init`

Create Docusaurus apps easily.

## Usage

Please see the [installation documentation](https://docusaurus.io/docs/installation).

## For maintainers

For Docusaurus maintainers, templates can be tested with:

```bash
cd `git rev-parse --show-toplevel` # Back to repo root
rm -rf test-website
yarn docusaurus-init init test-website classic
cd test-website
yarn start
```

Note: `test-website` is not part of the workspace and use packages from npm.

Use the following to test the templates against local packages:

```bash
cd `git rev-parse --show-toplevel` # Back to repo root
rm -rf test-website-in-workspace
yarn docusaurus-init init test-website-in-workspace classic
cd test-website-in-workspace
yarn start
```

Comment (Josh-Cena) [reply website/docs/typescript-support.md:23]: Um, the problem here is whether npm installs `@types/` dependencies transitively, so I do need to mock that process of running the whole `yarn install` process

Comment (slorber) [reply website/docs/typescript-support.md:23]: ah I see 😅

**Pull request merged**